akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserEvents.scala

```diff
@@ -11,44 +11,81 @@ import akka.cluster.sharding.typed.Murmur2NoEnvelopeMessageExtractor
 import akka.cluster.sharding.typed.scaladsl.ClusterSharding
 import akka.cluster.sharding.typed.scaladsl.Entity
 import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+import akka.actor.typed.delivery.ConsumerController
+import akka.actor.typed.delivery.ConsumerController.Start
+import akka.actor.typed.delivery.ConsumerController
+import akka.cluster.sharding.typed.delivery.ShardingConsumerController
+import akka.cluster.sharding.typed.ShardingEnvelope
+import akka.actor.typed.delivery.ConsumerController.SequencedMessage
+import akka.cluster.sharding.typed.Murmur2MessageExtractor
+import akka.actor.typed.delivery.ConsumerController.Confirmed
 
 object UserEvents {
 
-  val TypeKey: EntityTypeKey[UserEvents.Message] =
-    EntityTypeKey[UserEvents.Message]("user-processing")
+  val TypeKey: EntityTypeKey[SequencedMessage[UserEvents.Message]] =
+    EntityTypeKey[SequencedMessage[UserEvents.Message]]("user-processing")
 
   sealed trait Message extends CborSerializable {
     def userId: String
   }
   sealed trait UserEvent extends Message
-  case class UserAction(userId: String, description: String, replyTo: ActorRef[Done]) extends UserEvent
-  case class UserPurchase(userId: String, product: String, quantity: Long, priceInPence: Long, replyTo: ActorRef[Done])
+  case class UserAction(userId: String, description: String) extends UserEvent
+  case class UserPurchase(userId: String, product: String, quantity: Long, priceInPence: Long)
       extends UserEvent
 
   sealed trait UserQuery extends Message
   case class GetRunningTotal(userId: String, replyTo: ActorRef[RunningTotal]) extends UserQuery
 
+  final case class UserEventDelivery(message: Message, confirmTo: ActorRef[ConsumerController.Confirmed], seqNr: Long)
+
   case class RunningTotal(totalPurchases: Long, amountSpent: Long) extends CborSerializable
 
-  def apply(): Behavior[Message] = running(RunningTotal(0, 0))
+  def shardingInit(system: ActorSystem[_]): ActorRef[ShardingEnvelope[SequencedMessage[Message]]] = {
```

Comment:
reliable delivery mandated this not use a message extractor as it depends on ShardingEnvelope